test-support/helpers/utility.js

```diff
@@ -0,0 +1,10 @@
+export function removeWhiteSpace(str) {
+  return str.replace(/\s+/g, ' ');
+}
+
+export function nodesToTextArray($nodeList) {
```

Comment:
good points. will change `nodes` -> `elements`. 

for the empty case and nested case, this method is just for one level conversion, we don't handle any of those cases inside of the code.